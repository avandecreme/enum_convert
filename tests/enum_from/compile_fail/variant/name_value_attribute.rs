use enum_convert::EnumFrom;

enum Source {
    Unit,
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from = "Source::Unit"] // Should be #[enum_from(Source::Unit)]
    Unit,
}

fn main() {}
