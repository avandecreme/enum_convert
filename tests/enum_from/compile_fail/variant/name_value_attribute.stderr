error: expected either #[enum_from] (if there is no ambiguity) or a list of variants, for example #[enum_from(Source1::VariantA, Source2::VariantB)]
  --> tests/enum_from/compile_fail/variant/name_value_attribute.rs:10:5
   |
10 |     #[enum_from = "Source::Unit"] // Should be #[enum_from(Source::Unit)]
   |     ^
